### ISSUE TITLE
Cannot get the assets uploaded properly for a release

### ISSUE THREAD

**Issue by pstephenson02** [semantic-release/github#338]
Here is my semantic-release configuration in `package.json`:

```json
  "plugins": [
    "@semantic-release/commit-analyzer",
    "@semantic-release/release-notes-generator",
    [
      "@semantic-release/github",
      {
        "assets": [
          "out"
        ]
      }
    ]
  ]
```

Upon a commit with the message: `fix(ci): Message here` I get the following output:
```
❯❯ phils :: nextjs-blog git(main) 16:12  .\node_modules\.bin\semantic-release --no-ci
[4:12:26 PM] [semantic-release] » i  Running semantic-release version 17.3.4
[4:12:26 PM] [semantic-release] » √  Loaded plugin "verifyConditions" from "@semantic-release/npm"
[4:12:26 PM] [semantic-release] » √  Loaded plugin "verifyConditions" from "@semantic-release/github"
[4:12:26 PM] [semantic-release] » √  Loaded plugin "analyzeCommits" from "@semantic-release/commit-analyzer"
[4:12:26 PM] [semantic-release] » √  Loaded plugin "generateNotes" from "@semantic-release/release-notes-generator"
[4:12:26 PM] [semantic-release] » √  Loaded plugin "prepare" from "@semantic-release/npm"
[4:12:26 PM] [semantic-release] » √  Loaded plugin "publish" from "@semantic-release/npm"
[4:12:26 PM] [semantic-release] » √  Loaded plugin "publish" from "@semantic-release/github"
[4:12:26 PM] [semantic-release] » √  Loaded plugin "addChannel" from "@semantic-release/npm"
[4:12:26 PM] [semantic-release] » √  Loaded plugin "addChannel" from "@semantic-release/github"
[4:12:26 PM] [semantic-release] » √  Loaded plugin "success" from "@semantic-release/github"
[4:12:26 PM] [semantic-release] » √  Loaded plugin "fail" from "@semantic-release/github"
[4:12:32 PM] [semantic-release] » √  Run automated release from branch main on repository https://github.com/pstephenson02/nextjs-blog.git
[4:12:34 PM] [semantic-release] » √  Allowed to push to the Git repository
[4:12:34 PM] [semantic-release] » i  Start step "verifyConditions" of plugin "@semantic-release/npm"
[4:12:34 PM] [semantic-release] » √  Completed step "verifyConditions" of plugin "@semantic-release/npm"
[4:12:34 PM] [semantic-release] » i  Start step "verifyConditions" of plugin "@semantic-release/github"
[4:12:34 PM] [semantic-release] [@semantic-release/github] » i  Verify GitHub authentication
[4:12:34 PM] [semantic-release] » √  Completed step "verifyConditions" of plugin "@semantic-release/github"
[4:12:34 PM] [semantic-release] » i  Found git tag v0.1.4 associated with version 0.1.4 on branch main
[4:12:34 PM] [semantic-release] » i  Found 1 commits since last release
[4:12:34 PM] [semantic-release] » i  Start step "analyzeCommits" of plugin "@semantic-release/commit-analyzer"
[4:12:34 PM] [semantic-release] [@semantic-release/commit-analyzer] » i  Analyzing commit: fix(ci): Try a different glob
[4:12:34 PM] [semantic-release] [@semantic-release/commit-analyzer] » i  The release type for the commit is patch
[4:12:34 PM] [semantic-release] [@semantic-release/commit-analyzer] » i  Analysis of 1 commits complete: patch release
[4:12:34 PM] [semantic-release] » √  Completed step "analyzeCommits" of plugin "@semantic-release/commit-analyzer"
[4:12:34 PM] [semantic-release] » i  The next release version is 0.1.5
[4:12:34 PM] [semantic-release] » i  Start step "generateNotes" of plugin "@semantic-release/release-notes-generator"
[4:12:34 PM] [semantic-release] » √  Completed step "generateNotes" of plugin "@semantic-release/release-notes-generator"
[4:12:34 PM] [semantic-release] » i  Start step "prepare" of plugin "@semantic-release/npm"
[4:12:34 PM] [semantic-release] [@semantic-release/npm] » i  Write version 0.1.5 to package.json in C:\Users\phils\Code\pstephenson02\nextjs-blog
v0.1.5
[4:12:35 PM] [semantic-release] » √  Completed step "prepare" of plugin "@semantic-release/npm"
[4:12:44 PM] [semantic-release] » √  Created tag v0.1.5
[4:12:44 PM] [semantic-release] » i  Start step "publish" of plugin "@semantic-release/npm"
[4:12:44 PM] [semantic-release] [@semantic-release/npm] » i  Skip publishing to npm registry as package.json's private property is true
[4:12:44 PM] [semantic-release] » √  Completed step "publish" of plugin "@semantic-release/npm"
[4:12:44 PM] [semantic-release] » i  Start step "publish" of plugin "@semantic-release/github"
[4:12:45 PM] [semantic-release] [@semantic-release/github] » i  Published GitHub release: https://github.com/pstephenson02/nextjs-blog/releases/tag/v0.1.5
[4:12:45 PM] [semantic-release] » √  Completed step "publish" of plugin "@semantic-release/github"
[4:12:45 PM] [semantic-release] » i  Start step "success" of plugin "@semantic-release/github"
[4:12:47 PM] [semantic-release] » √  Completed step "success" of plugin "@semantic-release/github"
[4:12:47 PM] [semantic-release] » √  Published release 0.1.5 on default channel
```

![image](https://user-images.githubusercontent.com/6306934/105109601-b2a69d00-5a71-11eb-9ad7-8357a6c81044.png)

The release gets generated, but the assets (neither the .zip nor the .tar.gz) do not include the contents of the `out` directory.

I feel like I'm missing something really obvious here?

EDIT: Just to clarify, what I expect to happen is that all the files in the `out` directory get bundled into the .zip and .tar.gz files and uploaded with the release. I have also tried the glob `out/**`

**Comment by pstephenson02** [comment]
Just tried the following config under the assumption that maybe [this documentation](https://semantic-release.gitbook.io/semantic-release/usage/plugins#plugin-options-configuration) is out of date. The [reference here](https://github.com/semantic-release/github#assets) suggests that the `path` key is required.
```json
  "plugins": [
    "@semantic-release/commit-analyzer",
    "@semantic-release/release-notes-generator",
    [
      "@semantic-release/github",
      {
        "assets": [
          {
            "path": "out"
          }
        ]
      }
    ]
  ]
```

still no luck.

**Comment by gr2m** [comment]
can you try `"assets": "out/**"`? Can you run `ls out` before `semantic-release` to make sure the files exist?

**Comment by pstephenson02** [comment]
```
❯❯ phils :: nextjs-blog git(main) 11:58  ls out/

    Directory: C:\Users\phils\Code\pstephenson02\nextjs-blog\out

Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
d----           1/20/2021 11:52 AM                _next
-a---           1/20/2021 11:52 AM           2686 404.html
-a---           1/11/2021  2:56 PM          15086 favicon.ico
-a---           1/20/2021 11:52 AM           7305 index.html
-a---           1/11/2021  2:56 PM           1101 vercel.svg
```

```json
"@semantic-release/github",
{
  "assets": "out/**"
}
```

No luck 😢 

Could it have to do with the fact that I have: `"private": true,` in my package.json file?

**Comment by pstephenson02** [comment]
FWIW: Running: `semantic-release@17.3.4`

**Comment by pstephenson02** [comment]
I've discovered my issue finally. https://github.com/semantic-release/semantic-release/blob/master/docs/usage/configuration.md#configuration-file

> Note: When configuring via package.json, the configuration must be under the release property. However, when using a .releaserc or a release.config.js file, the configuration must be set without a release property.

I did not have my plugin configuration under the "release" property and hence it was not picking up my configuration at all. I wish this had been more clear, but obviously I did not read it carefully.

Closing this. Thanks
